[WGSL] WGSL::PhaseTimer::m_phaseName is uninitialized when dumpPhaseTimes is false

ddkilzer opened the pull request:
#### 2dc6cd386ad3ad98c9364315452ceb873a2e929d
<pre>
[WGSL] WGSL::PhaseTimer::m_phaseName is uninitialized when dumpPhaseTimes is false
<a href="https://bugs.webkit.org/show_bug.cgi?id=268254">https://bugs.webkit.org/show_bug.cgi?id=268254</a>
&lt;<a href="https://rdar.apple.com/121808602">rdar://121808602</a>&gt;

Reviewed by Darin Adler.

* Source/WebGPU/WGSL/PhaseTimer.h:
- Initialize m_phaseName to nullptr.

Canonical link: <a href="https://commits.webkit.org/273627@main">https://commits.webkit.org/273627@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b92aba15631f776e52598075e8793c62e8e503e1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36139 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15086 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38341 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38860 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32490 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17524 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12114 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31182 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36694 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12742 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32062 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11168 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11194 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40106 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32807 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32610 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/37130 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11388 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9275 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35216 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13112 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11865 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4673 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12225 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebGPU/WGSL/PhaseTimer.h:
```diff
@@ -100,7 +100,7 @@ class PhaseTimer {
 
 private:
-    const char* m_phaseName;
+    const char* m_phaseName { nullptr };
```
I guess this is OK. If the null phase name was ever appended to the vector, we’d have a problem later anyway. but this does no harm.

Commit:
[WGSL] WGSL::PhaseTimer::m_phaseName is uninitialized when dumpPhaseTimes is false
https://bugs.webkit.org/show_bug.cgi?id=268254
<rdar://121808602>

Reviewed by Darin Adler.

* Source/WebGPU/WGSL/PhaseTimer.h:
- Initialize m_phaseName to nullptr.

Canonical link: https://commits.webkit.org/273627@main
